src/main/java/com/google/api/codegen/viewmodel/FileHeaderView.java

```diff
@@ -40,6 +40,12 @@ public boolean hasVersion() {
     return version() != null && version().length() > 0;
   }
 
+  public abstract String generatorVersion();
+
+  public boolean hasGeneratorVersion() {
+    return generatorVersion() != null && generatorVersion().length() > 0;
```

Comment:
It's slightly more efficient to cache the value of `generatorVersion()` and then do these tests.